### PR TITLE
feat: support query(sql, object)

### PR DESCRIPTION
**Pull request by fangk**
close: #11

### CODE REVIEW

**Comment by fengmk2** [comment]
wait for https://github.com/ali-sdk/ali-rds/pull/13

**Comment by fengmk2** [comment]
rebase from master and test again should work.

File: lib/operator.js [hunk 1]
```diff
@@ -28,7 +28,18 @@ proto.escapeId = function (value, forbidQualified) {
 };
 
 proto.format = function (sql, values, stringifyObjects, timeZone) {
-  return SqlString.format(sql, values, stringifyObjects, timeZone);
+  // 不是数组, 而且是object的时候, 执行object替换;
```
Comment (fengmk2) [inline lib/operator.js]: 不要写中文。。。。全 en

File: lib/operator.js [hunk 2]
```diff
@@ -28,7 +28,18 @@ proto.escapeId = function (value, forbidQualified) {
...
+  if (!Array.isArray(values) && typeof values === 'object') {
```
Comment (fengmk2) [inline lib/operator.js]: 加上 `!values` 判断, null 也是 object

File: test/operator.test.js [hunk 3]
```diff
@@ -30,6 +30,19 @@ describe('operator.test.js', function () {
...
+      let op = new Operator();
+      assert.equal(op.format('SET name = ?', 'test'), 'SET name = \'test\'');
+    });
+
+    it('should get literal string by object', function () {
+      let op = new Operator();
+      assert.equal(op.format('SET dt = :now and name = :name and age = :age', {
+        now: op.literals.now,
+        name: 'test'
+      }), 'SET dt = now() and name = \'test\' and age = :age');
```
Comment (fengmk2) [inline test/operator.test.js]: 加上边界条件，null, undefined, 0, 1, 'foo', true, false 等。

**Comment by fengmk2** [comment]
commit author 不对

image

**Comment by fangk** [comment]
都改好了

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
2.6.0

**Pull request merged**